File: src/main/java/com/lcaohoanq/nocket/domain/avatar/Avatar.java
package com.lcaohoanq.nocket.domain.avatar;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.lcaohoanq.nocket.base.entity.BaseEntity;
import com.lcaohoanq.nocket.domain.user.User;
import com.lcaohoanq.nocket.metadata.MediaMeta;
import jakarta.persistence.AttributeOverride;
import jakarta.persistence.AttributeOverrides;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "avatars")
@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@AttributeOverrides({
    @AttributeOverride(name = "createdAt", column = @Column(name = "avatar_created_at")),
    @AttributeOverride(name = "updatedAt", column = @Column(name = "avatar_updated_at"))
})
public class Avatar extends BaseEntity {
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    private User user;
    
    @Embedded
    private MediaMeta mediaMeta;
    
}

File: server/src/main/kotlin/com/lcaohoanq/nocket/domain/avatar/Avatar.kt
package com.lcaohoanq.nocket.domain.avatar

import com.fasterxml.jackson.annotation.JsonBackReference
import com.lcaohoanq.nocket.base.entity.BaseEntity
import com.lcaohoanq.nocket.domain.user.User
import com.lcaohoanq.nocket.metadata.MediaMeta
import jakarta.persistence.*
import org.hibernate.envers.Audited

@Audited
@Entity
@Table(name = "avatars")
@AttributeOverrides(
    AttributeOverride(
        name = "createdAt",
        column = Column(name = "avatar_created_at")
    ),
    AttributeOverride(
        name = "updatedAt",
        column = Column(name = "avatar_updated_at")
    )
)
class Avatar : BaseEntity() {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    var user: User? = null
    
    @Embedded
    var mediaMeta: MediaMeta? = null
}
